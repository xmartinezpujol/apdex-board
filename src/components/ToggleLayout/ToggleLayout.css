.toggle-layout {
    position: relative;
    margin-left: 30px;
    font-size: 14px;
}

.toggle-layout:hover {
    cursor: pointer;
}

.toggle-layout label,
.toggle-layout:hover label:before,
.toggle-layout.active:hover label:before,
.toggle-layout  input[type="checkbox"]{
    pointer-events: none;
}

.toggle-layout  input[type="checkbox"] {
    position: relative;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    right: 5px;
    width: 15px;
    height: 15px;
    margin: 0;
    z-index: 1;
    opacity: 0;
}

.toggle-layout label:before {
    transition: all 0.3s ease;
    content: "";
    position: absolute;
    top: 2px;
    left: -5px;
    height: 13px;
    width: 13px;
    background-color: #FFF;
    border-radius: 4px;
    border: 3px solid white;
}

.toggle-layout:hover label:before {
    background-color: #CCC;
}

.toggle-layout.active:hover label:before {
    background-color: #FFF;
}

.toggle-layout.active label:before {
    background-color: #00BE33;
}

