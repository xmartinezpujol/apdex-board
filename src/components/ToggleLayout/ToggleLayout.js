import Component from './../Component';

import './ToggleLayout.css';

class ToggleLayout extends Component {
  constructor() {
    super();
    this.state = {
      layoutColumn: false,
    };
  }

  handleToggle() {
    this.state.layoutColumn = !this.state.layoutColumn;
    document.componentReg[0].update(this.state.layoutColumn);
    document.componentReg[this._id - document.currentState].update();
  }

  update() {
    document.getElementById(`toggle-${this._id}`).outerHTML =
      document.componentReg[this._id - document.currentState].render();
  }

  render() {
    const toggleWidth = this.state.layoutColumn ? 'active' : '';

    return `
      <div
          id="toggle-${this._id}"
          class="toggle-layout ${toggleWidth}"
          onClick="document.componentReg[${this._id - document.currentState}].handleToggle()"
        >
        <input
          name="layout-mode"
          type="checkbox"
          ${this.state.layoutColumn === true ? 'checked' : ''}
        />
        <label name="layout-mode ${toggleWidth}">
          ${this.state.layoutColumn === true ? 'Show as an awesome grid' : 'Show as list'}
        </label>
      </div>
   `;
  }
}

export default ToggleLayout;
