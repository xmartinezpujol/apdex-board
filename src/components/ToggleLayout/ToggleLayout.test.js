import ToggleLayout from './ToggleLayout';

const toggleLayout = new ToggleLayout();
document.body.innerHTML = toggleLayout.render();

describe('ToggleLayout', () => {
  it('starts unchecked', () => {
    expect(toggleLayout.state.layoutColumn).toBe(false);
  });

  it('renders correctly initial state', () => {
    expect(toggleLayout.render()).toMatchSnapshot();
  });

  // Toggle State
  it('changes layout mode on toggle', () => {
    document.componentReg[toggleLayout._id - document.currentState].handleToggle();
    expect(document.getElementById(`toggle-${toggleLayout._id}`).classList.contains('active')).toBe(true);
  });

  it('renders correctly the checked state', () => {
    expect(toggleLayout.render()).toMatchSnapshot();
  });
});

