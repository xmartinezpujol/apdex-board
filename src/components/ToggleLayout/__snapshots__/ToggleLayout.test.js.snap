// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleLayout renders correctly initial state 1`] = `
"
      <div
          id=\\"toggle-0\\"
          class=\\"toggle-layout \\"
          onClick=\\"document.componentReg[0].handleToggle()\\"
        >
        <input
          name=\\"layout-mode\\"
          type=\\"checkbox\\"
          
        />
        <label name=\\"layout-mode \\">
          Show as list
        </label>
      </div>
   "
`;

exports[`ToggleLayout renders correctly the checked state 1`] = `
"
      <div
          id=\\"toggle-0\\"
          class=\\"toggle-layout active\\"
          onClick=\\"document.componentReg[0].handleToggle()\\"
        >
        <input
          name=\\"layout-mode\\"
          type=\\"checkbox\\"
          checked
        />
        <label name=\\"layout-mode active\\">
          Show as an awesome grid
        </label>
      </div>
   "
`;
